File: sdk/navbar.go
package sdk

// NavbarProjectData is the sub struct returned by getNavbarHandler
type NavbarProjectData struct {
	Key             string `json:"key" db:"key"`
	Name            string `json:"name" db:"project_name"`
	Description     string `json:"description" db:"description"`
	ApplicationName string `json:"application_name,omitempty" db:"application_name"`
	WorkflowName    string `json:"workflow_name,omitempty" db:"workflow_name"`
	Type            string `json:"type,omitempty" db:"type"`
	Favorite        bool   `json:"favorite" db:"favorite"`
}

File: contrib/vscode-cds/src/models/navbar.ts
export interface NavbarData {
    projects: Array<NavbarProjectData>;
}

export interface NavbarProjectData {
    key: string;
    name: string;
    description: string;
    application_name: string;
    workflow_name: string;
    type: string;
    favorite: boolean;
}

export interface NavbarRecentData {
    project_key: string;
    name: string; // workflow name
}

export interface NavbarSearchItem {
    type: string;
    value: string;
    title: string;
    projectKey: string;
    favorite?: boolean;
}
